### PR TITLE
feat: options to customize cookie names

### PR DESCRIPTION
**Pull request by vtkac**
Hi, I do realize that these are very short-lived cookies, but I have a project where I have to follow a naming convention for all cookies, so it would be great if we could customize these too.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,7 +26,10 @@ function defaultGenerateStateFunction (request, callback) {
 
 function defaultCheckStateFunction (request, callback) {
   const state = request.query.state
-  const stateCookie = request.cookies['oauth2-redirect-state']
+  const stateCookie =
+    request.cookies[
+      this.redirectStateCookieName || DEFAULT_REDIRECT_STATE_COOKIE_NAME
```
Comment (Eomm) [inline index.js:31]: this is a potential security issue - we can't try to read both cookie

File: index.js [hunk 2]
```diff
@@ -116,10 +134,14 @@ function fastifyOauth2 (fastify, options, next) {
       tokenRequestParams = {},
       scope,
       generateStateFunction = defaultGenerateStateFunction,
-      checkStateFunction = defaultCheckStateFunction,
+      checkStateFunction = defaultCheckStateFunction.bind({
+        redirectStateCookieName: configured.redirectStateCookieName
```
Comment (Eomm) [inline index.js:138]:
```suggestion
        redirectStateCookieName: configured.redirectStateCookieName || DEFAULT_REDIRECT_STATE_COOKIE_NAME
```

**Review by Eomm — CHANGES_REQUESTED** [review]

**Review by Eomm — COMMENTED** [review]
We should update the Readme too

File: index.js (hunk 2 above)
Comment (vtkac) [reply index.js:138]: oh, ok, I see. Sure I can move that default value from `defaultCheckStateFunction` here, it should effectively do the same

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@Eomm PTAL

**Review by Eomm — APPROVED** [review]

**Pull request merged**